requirements.txt

```diff
@@ -1,5 +1,5 @@
 #
-# This file is autogenerated by pip-compile with Python 3.11
+# This file is autogenerated by pip-compile with Python 3.12
```

Comment:
Should we add a task to run pip-compile from inside the container in order to avoid this version drift?

Reply:
I was unsure about create a circular dependency here, where updating the dependencies requires first launching a container that installs the existing dependencies, then running a command on said container, then rebuilding the container...

For simplicity from an end-user point of view it would be better to have everything managed in the same way, and not have particular dependencies you need to install on your local machine yourself. I had a similar thought about the current need to install `tox` locally.

I can rework the `pip-compile` and `tox` commands to run on containers if need-be. What do you think?

Reply:
Once this is merged, I can open a PR to move them into the container for local dev. Thanks!

Reply:
See: https://github.com/cabotage/cabotage-app/pull/81